### PR TITLE
Factor out openstack_get_domain_id

### PR DESCRIPTION
**Pull request by tipabu**
##### SUMMARY
We previously had (nearly) the same code replicated across four different modules. This change pulls it out to a common function. This will make it easier for future modules to perform similar domain ID lookups.

##### ISSUE TYPE
- Refactor Pull Request

##### COMPONENT NAME
OpenStack cloud modules

##### ADDITIONAL INFORMATION
(None)

### CODE REVIEW

**Comment by cloudnull** [comment]
@tipabu  this needs to be rebased.

File: lib/ansible/modules/cloud/openstack/os_project.py [hunk 1]
```diff
@@ -99,7 +99,10 @@
 '''
 
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.openstack import openstack_full_argument_spec, openstack_module_kwargs, openstack_cloud_from_module
+from ansible.module_utils.openstack import (
```
Comment (cloudnull) [inline lib/ansible/modules/cloud/openstack/os_project.py:102]: While this works I'm not a fan of multiple imports on a single line. Not something I'd be willing to block the review over but something I'd suggest breaking out into multiple lines.

**Review by cloudnull — APPROVED** [review]
shipit

**Comment by sshnaidm** [comment]
Thanks for submitting patch for Openstack Ansible modules!
We moved Openstack Ansible modules to Openstack repositories.
Next patches should be submitted not with Ansible Github but with
Openstack Gerrit: https://review.opendev.org/#/q/project:openstack/ansible-collections-openstack
Please submit your code there from now.
Thanks for your contribution and sorry for inconvienience.

**Pull request closed without merging**